internal/services/baremetal/server.go

```diff
@@ -464,12 +485,35 @@ func ResourceServerRead(ctx context.Context, d *schema.ResourceData, m interface
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("failed to list server's private networks: %w", err))
 	}
+
 	pnRegion, err := server.Zone.Region()
 	if err != nil {
 		return diag.FromErr(err)
 	}
 	_ = d.Set("private_network", flattenPrivateNetworks(pnRegion, listPrivateNetworks.ServerPrivateNetworks))
 
+	privateNetworkIDs := make([]string, 0, len(listPrivateNetworks.ServerPrivateNetworks))
+	for _, pn := range listPrivateNetworks.ServerPrivateNetworks {
+		privateNetworkIDs = append(privateNetworkIDs, pn.PrivateNetworkID)
+	}
+
+	var allPrivateIPs []map[string]interface{}
```

Comment:
nit: Considering server will have at least 1 IP per connected private network. This should be allocated for the minimum size.